src/actor.rs

```diff
@@ -0,0 +1,196 @@
+use futures::prelude::*;
+use futures::stream::FuturesUnordered;
+use ractor::{
+    concurrency::Duration, Actor, ActorProcessingErr, ActorRef, BytesConvertable, RpcReplyPort,
+};
+use ractor_cluster::RactorClusterMessage;
+use serde::{Deserialize, Serialize};
+use threshold_crypto::{PublicKeySet, SecretKeyShare, Signature, SignatureShare};
+
+const MPC_RECOVERY_GROUP: &str = "mpc-recovery";
+
+type NodeId = u64;
+type Payload = Vec<u8>;
+
+#[derive(Debug, Clone, Serialize, Deserialize)]
+pub struct SignResponse {
+    node_id: NodeId,
+    sig_share: SignatureShare,
+}
+
+impl BytesConvertable for SignResponse {
+    fn into_bytes(self) -> Vec<u8> {
+        serde_json::to_vec(&self).unwrap()
+    }
+
+    fn from_bytes(bytes: Vec<u8>) -> Self {
+        serde_json::from_slice(&bytes).unwrap()
+    }
+}
+
+#[derive(Debug, Clone, Serialize, Deserialize)]
+pub struct SignatureResponse {
+    pub sig: Signature,
+}
+
+impl BytesConvertable for SignatureResponse {
+    fn into_bytes(self) -> Vec<u8> {
+        serde_json::to_vec(&self).unwrap()
+    }
+
+    fn from_bytes(bytes: Vec<u8>) -> Self {
+        serde_json::from_slice(&bytes).unwrap()
+    }
+}
+
+#[derive(RactorClusterMessage, Debug)]
+pub enum NodeMessage {
+    #[rpc]
+    NewRequest(Payload, RpcReplyPort<SignatureResponse>),
+    #[rpc]
+    SignRequest(Payload, RpcReplyPort<SignResponse>),
+}
+
+pub struct NodeActor;
+
+pub struct NodeActorState {
+    id: NodeId,
+    pk_set: PublicKeySet,
+    sk_share: SecretKeyShare,
+}
+
+#[async_trait::async_trait]
+impl Actor for NodeActor {
+    // An actor has a message type
+    type Msg = NodeMessage;
+    // and (optionally) internal state
+    type State = NodeActorState;
+    // Startup initialization args
+    type Arguments = (NodeId, PublicKeySet, SecretKeyShare);
```

Comment:
Why do we need to declare these types here?

Reply:
It is just a part of how ractor works. `Arguments` is what you need to pass to initialize the actor. In our case it is node id, public key set and the secret key share associated with this node.